conan/tools/gnu/autotools.py

```diff
@@ -0,0 +1,239 @@
+                triplet_args.append("--target=%s" % (target or self.target))
+
+        if pkg_config_paths:
+            pkg_env = {"PKG_CONFIG_PATH":
+                       [os.pathsep.join(get_abs_path(f, self._conanfile.install_folder)
+                                        for f in pkg_config_paths)]}
+        else:
+            # If we are using pkg_config generator automate the pcs location, otherwise it could
+            # read wrong files
+            pkg_env = {"PKG_CONFIG_PATH": [self._conanfile.install_folder]} \
+                if "pkg_config" in self._conanfile.generators else None
+        """
+        configure_dir = self._adjust_path(configure_dir)
+
+        """if self._conanfile.package_folder is not None:
+            if not args:
+                args = ["--prefix=%s" % self._conanfile.package_folder.replace("\\", "/")]
+            elif not self._is_flag_in_args("prefix", args):
+                args.append("--prefix=%s" % self._conanfile.package_folder.replace("\\", "/"))
+
+            all_flags = ["bindir", "sbindir", "libexecdir", "libdir", "includedir", "oldincludedir",
+                         "datarootdir"]
+            help_output = self._configure_help_output(configure_dir)
+            available_flags = [flag for flag in all_flags if "--%s" % flag in help_output]
+
+            if use_default_install_dirs:
+                for varname in ["bindir", "sbindir", "libexecdir"]:
+                    if self._valid_configure_flag(varname, args, available_flags):
+                        args.append("--%s=${prefix}/%s" % (varname, DEFAULT_BIN))
+                if self._valid_configure_flag("libdir", args, available_flags):
+                    args.append("--libdir=${prefix}/%s" % DEFAULT_LIB)
+                for varname in ["includedir", "oldincludedir"]:
+                    if self._valid_configure_flag(varname, args, available_flags):
+                        args.append("--%s=${prefix}/%s" % (varname, DEFAULT_INCLUDE))
+                if self._valid_configure_flag("datarootdir", args, available_flags):
+                    args.append("--datarootdir=${prefix}/%s" % DEFAULT_SHARE)
+
+        with environment_append(pkg_env):
+            with environment_append(vars or self.vars):
+                command = '%s/configure %s %s' % (configure_dir, args_to_string(args),
+                                                  " ".join(triplet_args))
+                self._conanfile.output.info("Calling:\n > %s" % command)
+                self._conanfile.run(command, win_bash=self._win_bash, subsystem=self.subsystem)"""
+
+        cmd = "bash -c 'source autotoolsdeps.sh && source autotools.sh && %s/configure'" % configure_dir
+        self._conanfile.output.info("Calling:\n > %s" % cmd)
+        self._conanfile.run(cmd, win_bash=self._win_bash, subsystem=self.subsystem)
+
+    def _configure_help_output(self, configure_path):
+        from six import StringIO  # Python 2 and 3 compatible
```

Comment:
for the new helper, I think we can avoid python 2 and six

Reply:
Agree, ongoing work.